Sources/MapboxMaps/Foundation/Camera/CameraManager.swift

```diff
@@ -174,122 +173,109 @@ public class CameraManager {
+                    completion: AnimationCompletion? = nil) -> CameraAnimatorProtocol? {
 
         guard let mapView = mapView else {
             return nil
         }
 
-        // Stop the `internalCameraAnimator` before beginning a `flyTo`
-        internalCameraAnimator?.stopAnimation()
-
-        guard let interpolator = FlyToInterpolator(from: mapView.cameraOptions,
-                                                   to: camera,
-                                                   size: mapView.bounds.size) else {
-            return nil
-        }
-
-        // If there was no duration specified, use a default
-        let time: TimeInterval = duration ?? interpolator.duration()
+        // Stop the `internalAnimator` before beginning a `flyTo`
+        internalAnimator?.stopAnimation()
 
-        // TODO: Consider timesteps based on the flyTo curve, for example, it would be beneficial to have a higher
-        // density of time steps at towards the start and end of the animation to avoid jiggling.
-        let timeSteps = stride(from: 0.0, through: 1.0, by: 0.025)
-        let keyTimes: [Double] = Array(timeSteps)
+        let flyToAnimator = FlyToAnimator(delegate: self)
+        mapView.cameraAnimatorsHashTable.add(flyToAnimator)
 
-        let animator = makeCameraAnimator(duration: time, curve: .linear) {
+        flyToAnimator.makeFlyToInterpolator(from: mapView.cameraOptions,
+                                             to: camera,
+                                             duration: duration,
+                                             screenFullSize: mapView.bounds.size)
 
-            UIView.animateKeyframes(withDuration: 0, delay: 0, options: []) {
-
-                for keyTime in keyTimes {
-                    let interpolatedCoordinate = interpolator.coordinate(at: keyTime)
-                    let interpolatedZoom = interpolator.zoom(at: keyTime)
-                    let interpolatedBearing = interpolator.bearing(at: keyTime)
-                    let interpolatedPitch = interpolator.pitch(at: keyTime)
-
-                    UIView.addKeyframe(withRelativeStartTime: keyTime, relativeDuration: 0.025) {
-                        self.mapView?.cameraView.centerCoordinate = interpolatedCoordinate
-                        self.mapView?.cameraView.zoom = CGFloat(interpolatedZoom)
-                        self.mapView?.cameraView.bearing = CGFloat(interpolatedBearing)
-                        self.mapView?.cameraView.pitch = CGFloat(interpolatedPitch)
-                    }
-                }
-            }
-        }
-
-        if let completion = completion {
-            animator.addCompletion(completion)
+        // Nil out the internalAnimator after `flyTo` finishes
+        flyToAnimator.addCompletion { [weak self](position) in
+            // Call the developer-provided completion (if present)
+            completion?(position)
+            self?.internalAnimator = nil
```

Comment:
Should probably reorder these so that things work if the developer chains animations

Reply:
Ah good catch, reordered.